topdown/cache/cache.go

```diff
@@ -67,24 +99,49 @@ type InterQueryCacheValue interface {
 // InterQueryCache defines the interface for the inter-query cache.
 type InterQueryCache interface {
 	Get(key ast.Value) (value InterQueryCacheValue, found bool)
-	Insert(key ast.Value, value InterQueryCacheValue) int
+	Insert(key ast.Value, value InterQueryCacheValue, expiresAt time.Time) int
 	Delete(key ast.Value)
 	UpdateConfig(config *Config)
 	Clone(value InterQueryCacheValue) (InterQueryCacheValue, error)
 }
 
 // NewInterQueryCache returns a new inter-query cache.
-func NewInterQueryCache(config *Config) InterQueryCache {
-	return &cache{
+func NewInterQueryCache(ctx context.Context, logger logging.Logger, config *Config) InterQueryCache {
```

Comment:
This is backwards incompatible. Instead consider adding `With*` methods for example `WithContext`. Also I don't think we need the logger here.

Reply:
Created a new factory `NewInterQueryCacheWithContext()`. We can plan to deprecate `NewInterQueryCache()` in favour of this one in the future. Not sure if we have any mechanism in place to marking deprecation in the repo.